File: backend/app/schemas/portfolio.py
from decimal import Decimal

from pydantic import BaseModel

from app.models.enums import DataSource
from app.schemas.trading import PositionOut


class AllocationSlice(BaseModel):
    symbol: str
    name: str
    value: Decimal
    percentage: Decimal


class PortfolioHistoryPoint(BaseModel):
    timestamp: int
    value: Decimal


class PortfolioSummary(BaseModel):
    data_source: DataSource
    total_portfolio_value: Decimal
    available_cash_balance: Decimal
    locked_balance: Decimal
    value_of_crypto_assets: Decimal
    total_unrealized_profit_loss: Decimal
    total_realized_profit_loss: Decimal
    total_profit_loss: Decimal
    total_profit_loss_pct: Decimal
    daily_profit_loss: Decimal
    daily_profit_loss_pct: Decimal
    currency: str
    allocation: list[AllocationSlice]
    positions: list[PositionOut]
    history: list[PortfolioHistoryPoint] = []

File: frontend-next/src/types/portfolio.ts
import type { DataSource, Position } from './trading'

export interface AllocationSlice {
  symbol: string
  name: string
  value: string
  percentage: string
}

export interface PortfolioHistoryPoint {
  timestamp: number
  value: string
}

export interface PortfolioSummary {
  data_source: DataSource
  total_portfolio_value: string
  available_cash_balance: string
  locked_balance: string
  value_of_crypto_assets: string
  total_unrealized_profit_loss: string
  total_realized_profit_loss: string
  total_profit_loss: string
  total_profit_loss_pct: string
  daily_profit_loss: string
  daily_profit_loss_pct: string
  currency: string
  allocation: AllocationSlice[]
  positions: Position[]
  history: PortfolioHistoryPoint[]
}
